frontend/src/components/IdentityProviderCreateForm.vue

```diff
@@ -0,0 +1,518 @@
+          </button>
+          <button
+            class="btn-primary"
+            :disabled="!allowUpdate"
+            @click="handleUpdateButtonClick"
+          >
+            {{ $t("common.update") }}
+          </button>
+        </template>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { cloneDeep, isEqual } from "lodash-es";
+import {
+  computed,
+  reactive,
+  defineEmits,
+  defineProps,
+  ref,
+  onMounted,
+} from "vue";
+import {
+  FieldMapping,
+  IdentityProvider,
+  IdentityProviderConfig,
+  IdentityProviderType,
+  OAuth2IdentityProviderConfig,
+  OIDCIdentityProviderConfig,
+} from "@/types/proto/v1/idp_service";
+import {
+  identityProviderTypeToString,
+  useIdentityProviderStore,
+} from "@/store/modules/idp";
+import { useActuatorStore } from "@/store";
+
+interface LocalState {
+  type: IdentityProviderType;
+}
+
+const props = defineProps<{
+  identityProviderName?: string;
+}>();
+
+const emit = defineEmits<{
+  (e: "delete", identityProvider: IdentityProvider): void;
+  (e: "cancel"): void;
+  (e: "confirm", identityProvider: IdentityProvider): void;
+}>();
+
+const identityProviderStore = useIdentityProviderStore();
+const state = reactive<LocalState>({
+  type: IdentityProviderType.OAUTH2,
+});
+const identityProvider = ref<IdentityProvider>(
+  IdentityProvider.fromPartial({})
+);
+const configForOAuth2 = ref<OAuth2IdentityProviderConfig>(
+  OAuth2IdentityProviderConfig.fromPartial({
+    fieldMapping: FieldMapping.fromPartial({}),
+  })
+);
+const scopesStringOfConfig = ref<string>("");
+const configForOIDC = ref<OIDCIdentityProviderConfig>(
+  OIDCIdentityProviderConfig.fromPartial({
+    fieldMapping: FieldMapping.fromPartial({}),
+  })
+);
+const identityProviderTypeList = [
+  IdentityProviderType.OAUTH2,
+  IdentityProviderType.OIDC,
+];
+
+const callbackUrl = computed(() => {
+  return `${
+    useActuatorStore().serverInfo?.externalUrl || window.origin
+  }/oauth/callback`;
+});
+
+const isCreating = computed(() => {
+  return !props.identityProviderName || props.identityProviderName === "";
+});
+
+const originIdentityProvider = computed(() => {
+  return identityProviderStore.getIdentityProviderByName(
+    props.identityProviderName || ""
+  );
+});
+
+const isFormCompleted = computed(() => {
+  if (
+    !identityProvider.value.name ||
+    !identityProvider.value.title ||
+    !identityProvider.value.domain
+  ) {
+    return false;
+  }
+
+  if (state.type === IdentityProviderType.OAUTH2) {
+    if (
+      !configForOAuth2.value.clientId ||
+      !configForOAuth2.value.clientSecret ||
+      !configForOAuth2.value.scopes ||
+      !configForOAuth2.value.authUrl ||
+      !configForOAuth2.value.tokenUrl ||
+      !configForOAuth2.value.userInfoUrl ||
+      !configForOAuth2.value.fieldMapping?.identifier ||
+      !configForOAuth2.value.fieldMapping?.displayName ||
+      !configForOAuth2.value.fieldMapping?.email
```

Comment:
@boojack aren't these two fields optional?

Reply:
will be updated in a followup PR

Reply:
FYI https://github.com/bytebase/bytebase/pull/4431